rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/action/TbCopyAttributesToEntityViewNode.java

```diff
@@ -0,0 +1,137 @@
+import org.thingsboard.rule.engine.api.RuleNode;
+import org.thingsboard.rule.engine.api.TbContext;
+import org.thingsboard.rule.engine.api.TbNode;
+import org.thingsboard.rule.engine.api.TbNodeConfiguration;
+import org.thingsboard.rule.engine.api.TbNodeException;
+import org.thingsboard.rule.engine.api.TbRelationTypes;
+import org.thingsboard.rule.engine.api.util.TbNodeUtils;
+import org.thingsboard.server.common.data.DataConstants;
+import org.thingsboard.server.common.data.EntityView;
+import org.thingsboard.server.common.data.kv.AttributeKvEntry;
+import org.thingsboard.server.common.data.kv.TsKvEntry;
+import org.thingsboard.server.common.data.plugin.ComponentType;
+import org.thingsboard.server.common.msg.TbMsg;
+import org.thingsboard.server.common.msg.session.SessionMsgType;
+import org.thingsboard.server.common.transport.adaptor.JsonConverter;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+import static org.thingsboard.rule.engine.api.util.DonAsynchron.withCallback;
+
+@Slf4j
+@RuleNode(
+        type = ComponentType.ACTION,
+        name = "copy attributes",
+        configClazz = EmptyNodeConfiguration.class,
+        nodeDescription = "Copy attributes from asset/device to entity view",
+        nodeDetails = "Copy attributes from asset/device to related entity view according to entity view configuration. \n " +
+                "Copy will be done only for attributes that are between start and end dates and according to attribute keys configuration",
+        uiResources = {"static/rulenode/rulenode-core-config.js"},
+        configDirective = "tbNodeEmptyConfig",
+        icon = "content_copy"
+)
+public class TbCopyAttributesToEntityViewNode implements TbNode {
+
+    EmptyNodeConfiguration config;
+
+    @Override
+    public void init(TbContext ctx, TbNodeConfiguration configuration) throws TbNodeException {
+        this.config = TbNodeUtils.convert(configuration, EmptyNodeConfiguration.class);
+    }
+
+    @Override
+    public void onMsg(TbContext ctx, TbMsg msg) throws ExecutionException, InterruptedException, TbNodeException {
+        if (msg.getType().equals(SessionMsgType.POST_ATTRIBUTES_REQUEST.name()) ||
+                msg.getType().equals(DataConstants.ATTRIBUTES_DELETED) ||
+                msg.getType().equals(DataConstants.ATTRIBUTES_UPDATED)) {
+            long now = System.currentTimeMillis();
+            String scope;
+            if (msg.getType().equals(SessionMsgType.POST_ATTRIBUTES_REQUEST.name())) {
+                scope = DataConstants.CLIENT_SCOPE;
+            } else {
+                scope = msg.getMetaData().getValue("scope");
+            }
+            ListenableFuture<List<EntityView>> entityViewsFuture =
+                    ctx.getEntityViewService().findEntityViewsByTenantIdAndEntityIdAsync(ctx.getTenantId(), msg.getOriginator());
```

Comment:
I assume no caching here? We need to add cache. This is a huge performance bottleneck. We need to also cache empty results. We also need to invalidate cache if one of the views was deleted/added/changed.